dotCMS/src/main/java/com/dotmarketing/quartz/job/ResetPermissionsJob.java

```diff
@@ -70,35 +74,23 @@ public ResetPermissionsJob() {
      *        Page, etc.
      */
 	public static void triggerJobImmediately (final Permissionable perm) {
-		final String randomID = UUID.randomUUID().toString();
-		final JobDataMap dataMap = new JobDataMap();
-
 		String userId = null;
-		
-		dataMap.put("permissionableId", perm.getPermissionId());
 
 		//TODO: For a major release, remove this logic and get userId as parameter
 		if (UtilMethods.isSet(HttpServletRequestThreadLocal.INSTANCE.getRequest())) {
 			userId = WebAPILocator.getUserWebAPI()
 					.getLoggedInUser(HttpServletRequestThreadLocal.INSTANCE.getRequest())
 					.getUserId();
 		}
-		dataMap.put("userId", userId);
-
-		final JobDetail jd = new JobDetail("ResetPermissionsJob-" + randomID, "dotcms_jobs", ResetPermissionsJob.class);
-		jd.setJobDataMap(dataMap);
-		jd.setDurability(false);
-		jd.setVolatility(false);
-		jd.setRequestsRecovery(true);
-		
-		final long startTime = System.currentTimeMillis();
-		final SimpleTrigger trigger = new SimpleTrigger("permissionsResetTrigger-"+randomID, "dotcms_triggers",  new Date(startTime));
+		final Map<String, Serializable> nextExecutionData = ImmutableMap
+				.of(
+						"permissionableId", perm.getPermissionId(),
+						"userId", userId);
 		
 		try {
+			DotStatefulJob.enqueueTrigger(nextExecutionData, ResetPermissionsJob.class);
 
-			final Scheduler sched = QuartzUtils.getSequentialScheduler();
-			sched.scheduleJob(jd, trigger);
-		} catch (SchedulerException e) {
+		} catch (Exception e) {

```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6190817)